lib/scope.js

```diff
@@ -304,4 +304,102 @@ module.exports = {
+      await scope.page.$$(`input[name="${ base64_name }"][value="True"]`),  // yesno/noyes checkbox
+      await scope.page.$$(`*[data-saveas="${ base64_name }"] input[value="True"]`),  // yesno/noyes checkbox showif
+      // Finds the group by looking for the first checkbox in the group using id, then works its way back out again
+      // 'none of the above' checkbox showif???
+      await scope.page.$$(`input[name="${ name_attr_B }"]`),  // checkbox
+      await scope.page.$$(`input[name="${ name_attr_R }"]`),  // checkbox alt
+      // await scope.page.$$(`*[data-saveas="${name_attr_B}"]`),  // checkbox showif?
+      // await scope.page.$$(`*[data-saveas="${name_attr_R}"]`),  // checkbox alt showif?
+
+      // Put these two last so they'll come last in the loop and won't override other options
+      await scope.page.$x(`//input[@id="${ base64_id_of_first_choice }"]/ancestor::fieldset//input[contains(@class,"danota-checkbox")]`),  // 'none of the above' checkbox
+      await scope.page.$$(`:not(button)[name="${ base64_name }"]`),  // others (not showif)
+    ];
+
+    for ( let attempt of all_attempts ) {
+      // There really should be only one anyway, but they're all lists
+      if ( attempt[0] ) {
+        let tag = await attempt[0].evaluate( elem => { return elem.tagName; });
+        return { handle: attempt[0], tag: tag };
+      }
+    }
+
+    // If we get out of the for loop, we didn't find the element
+    let err_message = `Cannot find the input with the variable name "${ var_name }"`;
+    if ( choice ) {
+      throw new ReferenceError( `${ err_message } and the value "${ choice_name }"` );
+    } else {
+      throw new ReferenceError( err_message );
+    }
+
+  },  // Ends scope.getInput()
+
+  setField: {
+    BUTTON: async function ( scope, { handle, set_to }) { await scope.tapElement( scope, handle ); },
+    TEXTAREA: async function ( scope, { handle, set_to }) {
+      await scope.setText( scope, { handle, set_to });
+      await scope.afterStep(scope, { waitForShowIf: true });
+    },
+    SELECT: async function ( scope, { handle, set_to }) {
+      await handle.select( set_to );
+      await scope.afterStep(scope, { waitForShowIf: true });
+    },
+    INPUT: async function ( scope, { handle, set_to }) {
+      /* Set value of some `input` element to the given value */
+
+      let type = await handle.evaluate( elem => { return elem.getAttribute('type'); });
+      if ( type === `radio` || type === `checkbox` ) {
+
+        let [label] = await handle.$x(`following-sibling::label`);
+        if ( type === `radio` ) { await label[ scope.activate ](); }
+        else { await scope.setCheckbox( scope, { label, set_to }); }
+
+      } else {
+        await scope.setText( scope, { handle, set_to });
+      }
+
+      await scope.afterStep(scope, { waitForShowIf: true });
+    },  // Ends scope.setField.INPUT()
+  },
+
+  setCheckbox: async function ( scope, { label, set_to }) {
```

Comment:
If we're only using `setCheckbox` in the `scope.js` file, what about starting the function name with an underscore (`_setCheckbox`) or in some other way indicating that this is a helper function that's not going to be used outside this file?

Reply:
We can definitely start doing that. I think it might be worth making this its own task and go through and do that for all functions that are just used internally.